### ISSUE TITLE
Error in `AST.modifyNodesByName`

### ISSUE THREAD

**Issue by mathisonian** [idyll-lang/idyll#486]
Take this markup:

````
[meta title:"Script Exec" description:"Short description of your project" /]

[Header
  fullWidth:true
  title:"Script Exec"
  subtitle:"Welcome to Idyll. Open index.idyll to start writing"
  author:"Your Name Here"
  authorLink:"https://idyll-lang.org"
  date:`(new Date()).toDateString()`
  background:"#222222"
  color:"#ffffff"
   /]

## Introduction


```exec:rstats
library(ggplot2)
ggplot(diamonds)  # if only the dataset is known.
```
````

and this AST code:

```
  AST.modifyNodesByName(ast, 'CodeHighlight', (node) => {
    console.log(node);
    return node;
  })
```

We'd expect that to find the codehighlight block, but it doesn't ever call the callback.

**Comment by mathisonian** [comment]
This appears just to be a capitalization issue, e.g. 

```
  AST.modifyNodesByName(ast, 'CodeHighlight', (node) => {
    console.log(node);
    return node;
  })
```

works

**Comment by mathisonian** [comment]
I think this should be updated to be case insensitive to match the rest of the API.

**Comment by rollaball** [comment]
fixed in #574
